### PR TITLE
handle file uploads

### PR DESCRIPTION
**Pull request by pulasthibandara**
Adds file upload support. Got some bugs with dependencies. Some regenerator runtime issue. Probably need a bable plugin :(

### CODE REVIEW

**Comment by pulasthibandara** [comment]
@hally9k got the `regeneratorRuntime` and `.mjs` issues fixed. but I'm not sure of the way i did it.

File: src/environment.js [hunk 1]
```diff
@@ -1,23 +1,52 @@
...
-        },
-        method: 'POST'
-    }).then((response: *): * => response.json())
-)
+type ReplacedFields = {
+    files: { [id: string]: File | Blob },
+    variables: { [id: string]: mixed },
+}
+
+const recReplace = (variables: Object): ReplacedFields => Object.entries(variables).reduce((acc: ReplacedFields, [key, value]: *): * => {
```
Comment (hally9k) [inline src/environment.js]: This is awesome BTW... What does `acc` mean though?

File: webpack.config.js [hunk 2]
```diff
@@ -30,7 +30,12 @@ let config = {
                 test: /\.(scss)$/,
                 include: path.resolve(__dirname, 'src'),
                 loaders: ['style-loader', 'css-loader', 'sass-loader']
-            }
+            },
+            {
+                test: /\.mjs$/,
+                include: /node_modules/,
+                type: 'javascript/auto',
```
Comment (hally9k) [inline webpack.config.js]: What's this type about, got a link to an article or something so I can understand it?

File: src/environment.js (hunk 1 above)
Comment (pulasthibandara) [reply src/environment.js]: `accumulator` just the default variable in `Array::reduce`

File: webpack.config.js (hunk 2 above)
Comment (pulasthibandara) [reply webpack.config.js]: I just copied from this

Comment (pulasthibandara) [reply webpack.config.js]: and its actually in the readme

**Comment by pulasthibandara** [comment]
Removed `react-relay-network-modern` dependency. It was easier writing it than dealing with all the shitty `regeneratorRuntime` and `.mjs` stuff

**Pull request merged**